src/jira.py

```diff
@@ -0,0 +1,144 @@
+import base64
+import re
+import requests
+import pprint
+import time
+
+from src.access import get_access_params
+from src.board import Issue
+
+
+class JiraBoard:
+
+    def __init__(self, project_key):
+        """
+        Create a Project storing all issues belonging to the provided project key
+
+        :param str project_key: The Jira project key e.g. "TEST"
+        """
+        self.url = get_access_params('jira')['options']['server']
+        self.token = get_access_params('jira')['api_token']
+        self.headers = {'X-Authentication-Token': self.token}
+        self.key = project_key
+        self.issues = dict()
+        self.api_call(0)  # Get information for all issues in the project
+
+    def api_call(self, start):
+        """
+        Make API calls until all results have been retrieved. Jira API responses can be paginated, defaulting to 50
+        results per page, so a new call has to be made until the total is reached.
+
+        :param start: The index in the results to start at. Always call this function with start=0
+        """
+
+        response = requests.get(self.url + 'search?jql=project=%s&startAt=%s' % (self.key, str(start)),
+                                headers=self.headers).json()
+
+        for i in response["issues"]:
+            self.issues[i["key"]] = JiraIssue(response=i)
+
+        if response['total'] >= start + response['maxResults']:  # There could be another page of results
+            self.api_call(start + response['maxResults'])
+
+
+class JiraIssue(Issue):
+
+    def __init__(self, key=None, response=None):
+        """
+        Create an Issue object from an issue key or from a portion of an API response
+
+        :param key: If specified, make an API call searching by this issue key
+        :param response: If specified, don't make a new API call but use this response from an earlier one
+        """
+        super().__init__()
+
+        self.url = get_access_params('jira')['options']['server']
+        self.token = base64.b64encode(get_access_params('jira')['api_token'].encode())
```

Comment:
I would do the base64 encoding in `get_access_params` instead (if it's true that this is always needed when we access Jira).

Reply:
good point